microcontrollers: start adding new boards arduino zero, esp32, esp826… (#114)
tinygo-org/tinygo-site: content/microcontrollers/esp8266.md

---
title: "ESP8266"
weight: 3
---

The [Espressif ESP8266]() is a tiny development board based on the Xtensa [ESP8266]() family of SoC.

## Interfaces

| Interface | Hardware Supported | TinyGo Support |
| --------- | ------------- | ----- |
| GPIO      | YES | YES |
| UART      | YES | YES |
| SPI      | YES | Not Yet |
| I2C      | YES | Not Yet |
| ADC      | YES | Not Yet |

> | PWM      | YES | Not Yet |

Reviewer: Perhaps add WiFi here too?

Author: Done.

## Machine Package Docs

[Documentation for the machine package for the ESP8266](../machine/esp8266)

## Flashing

### CLI Flashing on Linux

> You need to install the Espressif toolchain for Linux to use TinyGo with the ESP8266. Note that it is the same download as for the ESP32: 

Reviewer: The ESP8266 normally has a different toolchain, but in the case of TinyGo the ESP8266 also uses the ESP32 toolchain.

So you could for example say:

```suggestion
You need to install the same toolchain for the ESP8266 as is used for the ESP32 to use the ESP8266 with TinyGo:: 
```

Author: OK changed.

https://docs.espressif.com/projects/esp-idf/en/release-v3.0/get-started/linux-setup.html#standard-setup-of-toolchain-for-linux

In addition, you must install the `esptool` flashing tool:

https://github.com/espressif/esptool#easy-installation

Now you should be able to flash your board as follows:

- Plug your ESP8266 board into your computer's USB port.
- Build and flash your TinyGo code using the `tinygo flash` command. This command flashes the ESP8266 with the blinky1 example:

    ```

>     tinygo flash -target=esp8266 examples/blinky1

Reviewer: ```suggestion
    tinygo flash -target=esp8266 -port=/dev/ttyUSB0 examples/blinky1
```

(only for Linux)

Author: See above question.
